rest_lights.cpp

```diff
@@ -1973,101 +1973,113 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
-            taskRef.req.setDstEndpoint(0x01);
-
-            DBG_Printf(DBG_INFO, "Tuya debug 10: EP: %d ID : %s\n", ep, qPrintable(id));
-
-            if (targetOn)
-            {
-                data = QByteArray("\x01", 1);
-            }
-            else
-            {
-                data = QByteArray("\x00", 1);
-            }
-
-            ok = sendTuyaRequest(taskRef, TaskTuyaRequest, DP_TYPE_BOOL, button, data);
-
-            if (ok)
+    if (hasBri)
+    {
+        if (targetBri <= 0xFF)
+        {
+            quint16 bri = targetBri * 1000 / 254;
+            QByteArray data = QByteArray("\x00\x00",2);
+            data.append(static_cast<qint8>((bri >> 8) & 0xff));
+            data.append(static_cast<qint8>(bri & 0xff));
+            
+            if (R_GetProductId(taskRef.lightNode) == QLatin1String("Tuya_DIMSWITCH Earda Dimmer") ||
+                R_GetProductId(taskRef.lightNode) == QLatin1String("Tuya_DIMSWITCH EDM-1ZAA-EU"))
             {
-                QVariantMap rspItem;
-                QVariantMap rspItemState;
-                rspItemState[QString("/lights/%1/state/on").arg(id)] = targetOn;
-                rspItem["success"] = rspItemState;
-                rsp.list.append(rspItem);
-
-                //Not needed ?
-                //taskRef.lightNode->setValue(RStateOn, targetOn);
+                ok = sendTuyaRequest(taskRef, TaskTuyaRequest, DP_TYPE_VALUE, DP_IDENTIFIER_DIMMER_LEVEL_MODE2, data);
             }
             else
             {
-                rsp.list.append(errorToMap(ERR_INTERNAL_ERROR, QString("/lights/%1").arg(id), QString("Internal error, %1").arg(ERR_BRIDGE_BUSY)));
+                ok = sendTuyaRequest(taskRef, TaskTuyaRequest, DP_TYPE_VALUE, DP_IDENTIFIER_DIMMER_LEVEL_MODE1, data);
             }
         }
+        
+        if (ok)
+        {
+            QVariantMap rspItem;
+            QVariantMap rspItemState;
+            rspItemState[QString("/lights/%1/state/bri").arg(id)] = targetBri;
+            rspItem["success"] = rspItemState;
+            rsp.list.append(rspItem);
+        }
         else
         {
-            rsp.list.append(errorToMap(ERR_PARAMETER_NOT_AVAILABLE, QString("/lights/%1/state/on").arg(id), QString("parameter, not available")));
-            rsp.httpStatus = HttpStatusBadRequest;
-            return REQ_READY_SEND;
+            rsp.list.append(errorToMap(ERR_INTERNAL_ERROR, QString("/lights/%1").arg(id), QString("Internal error, %1").arg(ERR_BRIDGE_BUSY)));
         }
     }
-    else if (map.contains("alert"))
+
+    if (hasOn)
     {
-        if (map["alert"].type() == QVariant::String)
+        qint8 button = DP_IDENTIFIER_BUTTON_1;
+        QByteArray data;
+
+        //Retreive Fake endpoint, and change button value
+        uint8_t ep = taskRef.lightNode->haEndpoint().endpoint();
+        if (ep == 0x02) { button = DP_IDENTIFIER_BUTTON_2; }
+        if (ep == 0x03) { button = DP_IDENTIFIER_BUTTON_3; }

```

Comment:
```
if      (ep == 0x02) { button = DP_IDENTIFIER_BUTTON_2; }
else if (ep == 0x03) { button = DP_IDENTIFIER_BUTTON_3; }
else
{
    // what happens here? is 0x01 the only other valid endpoint?
}
```

Reply:
It keep the defaut value
`        qint8 button = DP_IDENTIFIER_BUTTON_1;`